### ISSUE TITLE
Chronos-2 for S&P500 Returns

### ISSUE THREAD

**Issue by pranavdhawann** [pranavdhawann/spring-2026-group4#75]
## Chronos-2: Zero-Shot Probabilistic Time Series Forecasting for S&P 500 Returns

A Chronos-2 based model for predicting 5-day forward returns of S&P 500 stocks using the
[FinMultiTime](https://huggingface.co/papers/2506.05019) dataset.

---

### Architecture

Chronos-2 is a pretrained probabilistic forecasting transformer (Amazon) that frames time
series prediction as a language modelling task over quantised value tokens.

| Component | Detail |
|---|---|
| **Backbone** | `amazon/chronos-t5-small` (Chronos-2 series) |
| **Input** | Univariate adjusted close price sequences, per ticker |
| **Quantisation** | 4096-bin uniform tokeniser over normalised values |
| **Output** | Predictive distribution over 5-day horizon (median as point estimate) |

The model is applied **zero-shot** — no fine-tuning on FinMultiTime — to establish a strong
pretrained time-series baseline. An optional fine-tuning stage adapts the model to S&P 500
return dynamics via LoRA on the encoder-decoder attention layers.

---

**Comment by pranavdhawann** [comment]
<img width="123" height="57" alt="Image" src="https://github.com/user-attachments/assets/185a9320-e476-407c-8c6e-9e57faf32f37" />
